pkg/scheduler/uid/uid.go

```diff
@@ -15,55 +15,46 @@ type UID struct {
 	ser   string
 }
 
-func New(group uint64, name string) *UID {
-	if name == "" {
-		name = uuid.New()
-	}
-	return &UID{
+func New(group uint64, name string) UID {
+	return UID{
```

Comment:
I moved generation to its own Generate method. I like to avoid optional parameters that need zero values passed because it makes the code harder to read. I could replace the default behavior with a fatal panic, however, since that's the idiomatic way to fail in a constructor due to an invalid usage error. If we need the constructor not to explode, the normal solution is to either validate before using it or have another static constructor that returns an error that can be handled by the caller.